src/components/SearchBar/index.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react'
+import {
+  StyledInput,
+  StyledForm
+} from './SearchBar.style'
+import { withRouter } from "react-router-dom";
+
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      query: this.props.match.params.term || ''
+    }
+  }
+
+  onSubmit (event) {
+    event.preventDefault();
+    const isFavorite = this.props.location.pathname.split('/')[1] === 'favorites';
```

Comment:
you should be able to use the props.match in order to match a path or url, review the [match](https://reacttraining.com/react-router/core/api/match) object